frontends/web/src/routes/account/account.tsx

```diff
@@ -61,11 +65,23 @@ export function Account({
   const [transactions, setTransactions] = useState<accountApi.TTransactions>();
   const [usesProxy, setUsesProxy] = useState<boolean>();
   const [insured, setInsured] = useState<boolean>(false);
+  const [uncoveredFunds, setUncoveredFunds] = useState<string[]>([]);
   const [stateCode, setStateCode] = useState<string>();
   const supportedExchanges = useLoad<SupportedExchanges>(getExchangeBuySupported(code), [code]);
 
   const account = accounts && accounts.find(acct => acct.code === code);
 
+  const checkUncoveredUTXOs = useCallback(async () => {
+    let uncoveredScripts: accountApi.ScriptType[] = [];
+    const utxos = await accountApi.getUTXOs(code);
+    utxos.forEach((utxo) => {
+      if (utxo.scriptType !== 'p2wpkh' && !uncoveredScripts.includes(utxo.scriptType)) {
+        uncoveredScripts.push(utxo.scriptType);
+      }
+    });
+    setUncoveredFunds(uncoveredScripts.map(script => getScriptName(script)));
+  }, [code]);
+
```

Comment:
This is "business logic" creeping into the frontend, not sure what I think of it. I know it is a bit easier to do it likes this sometimes, but ideally the frontend is solely for visualization of data delivered by the backend. 

What do you think of moving this to the backend w/ a separate API endpoint? I feel bad for asking because I see that it is more effort - how important is it that we stay consistent and keep the logic in the backend?

Reply:
Keeping as is, since this is consistent with #2239. Will keep in mind to move all the logic to the backend in the future, unless we find a good reason for not doing it